R/utils.R

```diff
@@ -22,3 +22,34 @@ description_file_attribute <- function(pkg_path, attribute) {
   as.character(read.dcf(file.path(pkg_path, "DESCRIPTION"))[1, attribute])
 }
 
+`commit_to_gh_pages!` <- function(directory, dir) {
+  ## A little bit of git magic
+
+  ## Now that we have the docs, we need to transfer them to the gh-pages
+  ## branch of the repo, commit and push.
+  ## Switch to gh-pages branch
+  cur_branch <- system('git rev-parse --abbrev-ref HEAD', intern = TRUE)
+  on.exit({
+    system(paste('git checkout', cur_branch))
+  })
+
+  st <- system2('git', 'checkout gh-pages')
+  if (st == 1) system2('git', 'checkout -b gh-pages')
+
+  ## Copy the docs to the repo folder, and push them upstream
+  system(paste0('cp -rf ', dir, "/* ", directory))
```

Comment:
It's a really quick and beneficial refactor to use the R interface rather than `system` calls.